packages/client/src/components/GameBoard.tsx

```
@@ -211,6 +212,189 @@ export const GameBoard: React.FC = () => {
+    });
+    return () => {
+      offGhost?.();
+      offClear?.();
+      offDragging?.();
+      offDragEnd?.();
+    };
+  }, [room]);
+
+  // Ghost staging only reflects the current defender. Drop stale entries whenever the
+  // turn or active attack composition changes.
+  useEffect(() => {
```

ESLint

[error] React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?

(react-hooks/rules-of-hooks)